### DISCUSSION TITLE
Maximum number of updates for WriteRelationships?

### DISCUSSION THREAD

**Discussion by theronic** [authzed/spicedb#1922] [Q&A]
What is the upper limit for number updates in a call to WriteRelationships? WriteRelationships commits are atomic and support Preconditions. The [docs for flags](https://authzed.com/docs/spicedb/concepts/commands) states that the default number of updates or preconditions is 1000. Is this an upper limit?

I am using Java gRPC client.

**Comment by vroldanbet** [comment]
Yep, `--write-relationships-max-updates-per-call ` is the flag that controls that, and the default is 1000.

**Reply by theronic** [reply]
OK, but what's the limit? How high can it go?

**Reply by vroldanbet** [reply]
Cranking it up will put more strain on your datastore (e.g. Posgres, MySQL...), affecting the latency and even availability of your database. It depends on the database, the capacity of said database, etc.

So I guess the answer is "it depends". You can increase it progressively to set it to a point where you think the tradeoff is acceptable.
